src/Config/ConfigWriter.php

```diff
@@ -2,217 +2,48 @@
-        }
-
-        return $result;
-    }
-
-    protected function parseContentValue($contents, $path, $value)
-    {
-        $result = $contents;
-        $items = explode('.', $path);
-        $key = array_pop($items);
-        $replaceValue = $this->writeValueToPhp($value);
-
-        $count = 0;
-        $patterns = [];
-        $patterns[] = $this->buildStringExpression($key, $items);
-        $patterns[] = $this->buildStringExpression($key, $items, '"');
-        $patterns[] = $this->buildConstantExpression($key, $items);
-        $patterns[] = $this->buildArrayExpression($key, $items);
-
-        foreach ($patterns as $pattern) {
-            $result = preg_replace($pattern, '${1}${2}'.$replaceValue, $result, 1, $count);
-
-            if ($count > 0) {
-                break;
-            }
-        }
-
-        return $result;
-    }
-
-    protected function writeValueToPhp($value)
-    {
-        if (is_string($value) && strpos($value, "'") === false) {
-            $replaceValue = "'".$value."'";
-        }
-        elseif (is_string($value) && strpos($value, '"') === false) {
-            $replaceValue = '"'.$value.'"';
-        }
-        elseif (is_bool($value)) {
-            $replaceValue = ($value ? 'true' : 'false');
-        }
-        elseif (is_null($value)) {
-            $replaceValue = 'null';
-        }
-        elseif (is_array($value) && count($value) === count($value, COUNT_RECURSIVE)) {
-            $replaceValue = $this->writeArrayToPhp($value);
-        }
-        else {
-            $replaceValue = $value;
-        }
-
-        $replaceValue = str_replace('$', '\$', $replaceValue);
-
-        return $replaceValue;
-    }
-
-    protected function writeArrayToPhp($array)
+    public function toFile(string $filePath, array $newValues): string
     {
-        $result = [];
-
-        foreach ($array as $value) {
-            if (!is_array($value)) {
-                $result[] = $this->writeValueToPhp($value);
-            }
-        }
-
-        return '['.implode(', ', $result).']';
+        $arrayFile = ArrayFile::open($filePath)->set($newValues);
+        $arrayFile->write();
+        return $arrayFile->render();
     }
 
-    protected function buildStringExpression($targetKey, $arrayItems = [], $quoteChar = "'")
+    public function toContent(string $contents, $newValues): string
     {
-        $expression = [];
-
-        // Opening expression for array items ($1)
-        $expression[] = $this->buildArrayOpeningExpression($arrayItems);
-
-        // The target key opening
-        $expression[] = '([\'|"]'.$targetKey.'[\'|"]\s*=>\s*)['.$quoteChar.']';
-
-        // The target value to be replaced ($2)
-        $expression[] = '([^'.$quoteChar.']*)';
-
-        // The target key closure
-        $expression[] = '['.$quoteChar.']';
-
-        return '/' . implode('', $expression) . '/';
-    }
+        $parser = (new ParserFactory)->create(ParserFactory::PREFER_PHP7, $lexer = new Emulative([
```

Comment:
@jaxwilko could you make sure the definition of `$lexer` happens on a separate line above? It makes it easier for debuggers to follow.